policy-controller/grpc/src/outbound.rs

```diff
@@ -170,15 +180,23 @@ where
             .await
             .map_err(|e| tonic::Status::internal(format!("lookup failed: {e}")))?
             .ok_or_else(|| tonic::Status::not_found("unknown server"))?;
-        Ok(tonic::Response::new(response_stream(drain, rx)))
+        Ok(tonic::Response::new(response_stream(
+            drain,
+            rx,
+            self.allow_l5d_request_headers,
```

Comment:
This seems like a potentially awkward way of implementing this if we ever care to make this controlled at a finer grain than process-wide.

May I suggest that we drop support for header control from this PR and add it separately? That will reduce the surface of this PR a bit, I think, and let us be a little more considerate about that UX, while still landing all of the P0 functionality.